server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -278,10 +277,11 @@ public ShardSearchRequest(StreamInput in) throws IOException {
         numberOfShards = in.readVInt();
         scroll = in.readOptionalWriteable(Scroll::new);
         source = in.readOptionalWriteable(SearchSourceBuilder::new);
-        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)) {
-            rankQueryBuilders = in.readNamedWriteableList(QueryBuilder.class);
-        } else {
-            rankQueryBuilders = List.of();
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0) && in.getTransportVersion().before(TransportVersion.V_8_500_004)) {
+            List<QueryBuilder> rankQueryBuilders = in.readNamedWriteableList(QueryBuilder.class);
+            for (QueryBuilder queryBuilder : rankQueryBuilders) {
+                source.queries().add(new SearchQueryWrapperBuilder(queryBuilder));
```

Comment:
this is some confusing bit to me that I need to verify: we were previously serializing the separate queries as part of shard search request. they were the query + each knn query. Now that is supported upstream by SearchSourceBuilder as it supports multiple queries, but is knn also part of it and if so when does it become part of it? Do we have a bwc test that covers this scenario cross version?

Reply:
I will try to add more tests for this. If knn searches are specified, we know that we are doing a dfs-query-then-fetch type of search. This means that after the dfs phase returns, we rewrite the shard requests for knn searches to be queries. They are applied in the rewrite method of `DfsQueryPhase`. The knn searches are then removed from the shard requests which is how it was before as well.

Reply:
BWC for 8.8 covered this in multiple tests (a couple hundred failed :) ) and caught a bug here (the one that took a long time), so I would say coverage on this is good.